### PR TITLE
docs: Update the client certificates example to a working config

### LINKED ISSUE #1312: Errors from certificate configuration example in guidebook
The example at [](https://quinn-rs.github.io/quinn/quinn/certificate.html#configuring-certificates) doesn't seem to compile. 

Problem one is just an extra parenthesis: 
```rust
let mut crypto = rustls::ClientConfig::builder()
    .with_safe_defaults()
    .with_single_cert(vec![cert]), key)?  // <---- [cert]) should just be [cert]
    .with_no_client_auth();
let client_config = ClientConfig::new(Arc::new(crypto));
```

Fixing that leads to the following error:

```rust
error[E0599]: no method named `with_single_cert` found for struct `ConfigBuilder<rustls::ClientConfig, WantsVerifier>` in the current scope
  --> networking\src\lib.rs:30:10
   |
30 |         .with_single_cert(vec![cert], key)?
   |          ^^^^^^^^^^^^^^^^ method not found in `ConfigBuilder<rustls::ClientConfig, WantsVerifier>`
   |
   = note: the method was found for
           - `ConfigBuilder<rustls::ClientConfig, WantsTransparencyPolicyOrClientCert>`
           - `ConfigBuilder<rustls::ClientConfig, WantsClientCert>`
           - `ConfigBuilder<rustls::ServerConfig, WantsServerCert>`
```

Imo, it could also be clearer that this example isn't really following from the LetsEncrypt docs immediately above this section.

### PR DESCRIPTION
**Pull request by jihiggins**
Closes #1312

### CODE REVIEW

File: docs/book/src/quinn/certificate.md [hunk 1]
```diff
@@ -129,14 +129,10 @@ This is the only thing you need to do for your server to be secured.
 **Configure Client**
 
 ```rust
-let mut crypto = rustls::ClientConfig::builder()
-    .with_safe_defaults()
-    .with_single_cert(vec![cert]), key)?
-    .with_no_client_auth();
-let client_config = ClientConfig::new(Arc::new(crypto));
+let client_config = ClientConfig::with_native_roots();
```
Comment (djc) [inline docs/book/src/quinn/certificate.md:132]: This only works with hyper-rustls, I think? rustls itself doesn't supply `ClientConfig::with_native_roots()`. Instead, we should probably have some simplified version of what that code uses internally:

```rust
let mut roots = rustls::RootCertStore::empty();
        let mut valid_count = 0;
        let mut invalid_count = 0;

        for cert in rustls_native_certs::load_native_certs().expect("could not load platform certs")
        {
            let cert = rustls::Certificate(cert.0);
            match roots.add(&cert) {
                Ok(_) => valid_count += 1,
                Err(err) => {
                    crate::log::trace!("invalid cert der {:?}", cert.0);
                    crate::log::debug!("certificate parsing failed: {:?}", err);
                    invalid_count += 1
                }
            }
        }
```

Comment (Ralith) [reply docs/book/src/quinn/certificate.md:132]: I believe this is `quinn::ClientConfig`, which does have such a helper, not `rustls::ClientConfig`.

Comment (djc) [reply docs/book/src/quinn/certificate.md:132]: Ah, sorry, ok.

File: docs/book/src/quinn/certificate.md [hunk 2]
```diff
@@ -129,14 +129,10 @@ This is the only thing you need to do for your server to be secured.
...
 ```
 
-This is the only thing you need to do for your client to trust a server certificate. 
+This is the only thing you need to do for your client to trust a non-self-signed server certificate.
```
Comment (Ralith) [inline docs/book/src/quinn/certificate.md:135]:
```suggestion
This is the only thing you need to do for your client to trust a server certificate signed by a conventional certificate authority. 
```

**Review by Ralith — APPROVED** [review]
Thanks!

**Pull request merged**